### PR TITLE
Add missing settings to plugin allowed list

### LINKED ISSUE #1581: [BUG] Documented audit log settings do not work
**Describe the bug**
I'm trying to tweak the audit log settings, but I'm gettings this:

relevant snippet of opensearch.yml:
```
plugins:
  security:
    audit:
      type: internal_opensearch
      config:
        enable_transport: false
```

```
java.lang.IllegalArgumentException: unknown setting [plugins.security.audit.config.enable_transport] did you mean any of [plugins.security.audit.config.enable_ssl, plugins.security.audit.config.pemkey_password, plugins.security.audit.config.enable_ssl_client_auth, plugins.security.audit.config.password, plugins.security.audit.config.username, plugins.security.audit.config.cert_alias, plugins.security.audit.config.pemcert_content, plugins.security.audit.config.pemkey_content]?
```

**To Reproduce**
Add audit settings as shown above

**Expected behavior**
I would expect the server to start and the documented audit settings to work

**Plugins**
All defaults 

**Screenshots**

**Host/Environment (please complete the following information):**
Default docker image, version 1.2.4

**Additional context**

### PR DESCRIPTION
**Pull request by peternied**
### Description

As settings have migrated from `opendistro_security` -> `plugins.security`, there were missed settings for the audit filter entries.  This caused `unknown setting` errors when starting OpenSearch, and these settings values were not being applied correctly affecting the runtime behavior of the audit log filters.

Created FilterEntries enum, as the filter keys/namespaced-keys are used
in more than once spot if we add a new filter key, we need a way to know
if the entries were updated in the otherside of the codebase where the
defaults are used.  In the settings configuration, if a new enum value
is added and not included a runtime exception will fire on startup,
ensuring new settings are always included.

This could be better where the enum defines the default values and how
the settings are extracted from json, but that would be a considerable
larger scope with the inclusion of generics.

Signed-off-by: Peter Nied <petern@amazon.com>

### Issues Resolved
* Closes #1581

### Check List
- [x] New functionality includes testing
- [ ] ~New functionality has been documented~
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java [hunk 1]
```diff
@@ -933,6 +935,7 @@ public List<Setting<?>> getSettings() {
             settings.add(Setting.boolSetting(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_RESOLVE_INDICES, true, Property.NodeScope, Property.Filtered));
             settings.add(Setting.boolSetting(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_ENABLE_REST, true, Property.NodeScope, Property.Filtered));
             settings.add(Setting.boolSetting(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_ENABLE_TRANSPORT, true, Property.NodeScope, Property.Filtered));
+            settings.add(Setting.boolSetting("plugins.security.audit.config.enable_transport", true, Property.NodeScope, Property.Filtered));
```
Comment (DarshitChanpura) [inline src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java:938]: `"plugins.security.audit.config.enable_transport"` should it be moved to ConfigConstants?

Comment (peternied) [reply src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java:938]: Good catch, this is covered as part of the section of code at lines 950

**Review by DarshitChanpura — APPROVED** [review]

**Comment by peternied** [comment]
Moving back to draft until all test cases have been fix

File: src/main/java/org/opensearch/security/auditlog/config/AuditConfig.java [hunk 2]
```diff
@@ -130,9 +131,9 @@ public static AuditConfig from(final Settings settings) {
      */
     @JsonInclude(JsonInclude.Include.NON_NULL)
     public static class Filter {
+        private static Set<String> FIELDS = DefaultObjectMapper.getFields(Filter.class);
```
Comment (peternied) [inline src/main/java/org/opensearch/security/auditlog/config/AuditConfig.java:134]: After adding changes to the `.from` method, this made the order of these static fields was incorrect causing static constructors to fail.  A great reminder to use lazy loading avoiding static order of operations management.

**Comment by peternied** [comment]
@opensearch-project/security Could I get a review on this pull request?

**Review by DarshitChanpura — APPROVED** [review]

**Comment by cliu123** [comment]
@peternied Could you please explain what the root cause of the issue https://github.com/opensearch-project/security/issues/1581 is? That'll help me understand how this PR fixes the issue.

**Comment by cliu123** [comment]
@peternied Could you please explain what the root cause of the issue https://github.com/opensearch-project/security/issues/1581 is? That'll help me understand how this PR fixes the issue.

**Comment by cliu123** [comment]
@peternied Please merge when you are ready.

**Comment by peternied** [comment]
There is an issue with the main branch build process, I am going to hold off on merging until it has been resolved

**Pull request merged**